Updating error message for misleading error in add product attribute code

aman3103 opened the pull request:
### Description
Updating validate_code error message to more understandable form i.e initially it was "Please use only letters (a-z), numbers (0-9) or underscore (_) in this field, and the first character should be a letter" now it is "Please use only lowercase letters (a-z), numbers (0-9) or underscore (_) in this field, and the first character should be a letter"

### Manual testing scenarios
1. Navigate to Store->Attributes->Product
2. Add new attribute
3. Set Attribute Code to Foo_Bar
4. Click Save

### Fixed Issues (if relevant)
1. magento/magento2#17754: Misleading error in Add Product Attribute screen

aman3103 committed:
updating code to handle misleading error in add product attribute screen by make error message more genric

aman3103 committed:
updating code to handle misleading error in add product attribute screen by make error message more genric

magento-cicd2 commented:
[![CLA assistant check](https://cla.dev.magento.com/pull/badge/not_signed)](https://cla.dev.magento.com/magento/magento2?pullRequest=17800) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dev.magento.com/magento/magento2?pullRequest=17800) before we can accept your contribution.<br/><hr/>**ck-aagarwal** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).

magento-engcom-team commented:
Hi @aman3103. Thank you for your contribution 
Here is some useful tips how you can test your changes using Magento test environment.
Add the comment under your pull request to deploy test or vanilla Magento instance:
- `@magento-engcom-team give me test instance` - deploy test instance based on PR changes
- `@magento-engcom-team give me {$VERSION} instance` - deploy vanilla Magento instance

For more details, please, review the [Magento Contributor Assistant](https://devdocs.magento.com/guides/v2.2/contributor-guide/contributing.html#contributor-assist) documentation

osrecio reviewed (approved):
Thanks @aman3103  for your contribution. We will process your PR ASAP.

aman3103 committed:
making changes for Upload category image with same image name issue #17661

aman3103 committed:
Revert "making changes for Upload category image with same image name issue #17661"

This reverts commit 9a15e1f99ec742b96572285a6dd4dfb118c11d94.

aman3103 committed:
Merge remote-tracking branch 'upstream/2.3-develop' into 2.3-develop

aman3103 commented:
@magento-engcom-team : May I know why Travis is throwing an error in this PR?  I am not able to understand the error.

aman3103 commented:
@sidolov 
I have only one github account setup. 
So I need to make some commit first or can just try to push the branch forcefully?

slavvka commented:
@aman3103 Please add all your emails you used for commits in this PR to your GitHub profile (currently they are gray since they aren't tied up to any GitHub profile). Or you could force-push your fix under your primary email that is tied to your profile.

aman3103 commented:
> @aman3103 Please add all your emails you used for commits in this PR to your GitHub profile (currently they are gray since they aren't tied up to any GitHub profile). Or you could force-push your fix under your primary email that is tied to your profile.

I have pushed the fix forcefully already, thanks. Is there anything I have to do?

slavvka commented:
@aman3103 Are you sure that email `a*.a*al@ch*th.com` used in all your commits added to your profile (User/Settings/Email section)?

aman3103 commented:
> @aman3103 Are you sure that email `a*.a*al@ch*th.com` used in all your commits added to your profile (User/Settings/Email section)?

This email is not in working state, as I left that company. So I have forced pushed the changes with my primary email.

slavvka commented:
@aman3103 Thank you but there's another commit with the other email `a*al@w18*f1.local`

aman3103 commented:
> @aman3103 Thank you but there's another commit with the other email `a*al@w18*f1.local`

Sorry I think when I did the last commit it automatically took the email from my local git user as no user and email was setup on my system. Please let me know how to solve this one. 
Thanks.

slavvka commented:
@aman3103 please either add this email to you GitHub profile or squash and force-push your branch. Please make all commits to GitHub with the name and email that is registered in GitHub account

aman3103 commented:
> @aman3103 please either add this email to you GitHub profile or squash and force-push your branch. Please make all commits to GitHub with the name and email that is registered in GitHub account



> @aman3103 please either add this email to you GitHub profile or squash and force-push your branch. Please make all commits to GitHub with the name and email that is registered in GitHub account

Sure, I will take care about the commits in future. But I cannot add the email as that email you have mentioned was the fake which added by my system when committing based on my local username.
When I try to squash I am getting this error:
error: commit b6a899f0a0ef8fd45d94051f6267b6f95cda4fb3 is a merge but no -m option was given.
Could not apply b6a899f0a0e... 
I am doing something like this:
pick b6a899f0a0e 
squash 19e39084263 making commit to push branch forcefully

Is there anything wrong in this or anything wrong with my last to last commit?

aman3103 commented:
> @aman3103 you could create a patch (diff) based on your changes, reset your branch to the state before you made any changes, merge the mainline code and push the commit with all your changes.

I have removed the old commit as I have created that one just to force push the branch, don't know why travis is throwing the error. Is it because I have removed the commit?

magento-engcom-team commented:
Hi @slavvka, thank you for the review.
[ENGCOM-3165](https://jira.corp.magento.com/browse/ENGCOM-3165) has been created to process this Pull Request

magento-engcom-team commented:
Hi @aman3103. Thank you for your contribution. 
We will aim to release these changes as part of 2.3.1. 
Please check the release notes for final confirmation.

aman3103 commented:
> Hi @aman3103. Thank you for your contribution.
> We will aim to release these changes as part of 2.3.1.
> Please check the release notes for final confirmation.

Thanks. Release note seems fine.
